src/lambda_vm/lambda_vm.ml

```diff
@@ -1,29 +1,39 @@
 module Ast = Ast
 module Gas = Gas
+let wrap compute =
+  let wrap_error error = `Runtime_error error |> Result.error in
+  try compute () |> Result.map_error (fun error -> `Internal_error error) with
+  | Checks.Out_of_stack -> wrap_error `Out_of_stack
+  | Gas.Out_of_gas -> wrap_error `Out_of_gas
+module Error_set = struct
+  type 'error t =
+    [ `Internal_error of 'error
+    | `Runtime_error  of [`Out_of_gas | `Out_of_stack] ]
```

Comment:
Just use normal variants, also this additional module, only adds an additional level of indirection, I need to go up and down multiple times to understand what `error Error_set.t` meant.

This is not even reducing the code size